giraph-core/src/main/java/org/apache/giraph/utils/UnsafeReads.java

```diff
@@ -18,27 +18,24 @@
 
 package org.apache.giraph.utils;
 
+import com.esotericsoftware.kryo.io.Input;
+
 import java.io.IOException;
 import java.io.UTFDataFormatException;
 
 /**
  * Byte array input stream that uses Unsafe methods to deserialize
  * much faster
  */
-public abstract class UnsafeReads implements ExtendedDataInput {
-
-  /** Buffer length */
-  protected int bufLength;
-  /** Position in the buffer */
-  protected long pos = 0;
+public abstract class UnsafeReads extends Input implements ExtendedDataInput {
```

Comment:
Can we add some unit tests for UnsafeReads/UnsafeByteArrayInputStream just like with UnsafeByteArrayOutputStream?

Reply:
TestUnsafeByteArrayOutputStream tests both input and output streams. I have added more testing in next PR through kryo serialization.